src/server.rs

```diff
@@ -366,7 +371,11 @@ impl Server {
             .data_sent(messages.len(), self.process_id, self.address.id);
 
         match write_all_half(&mut self.write, messages).await {
-            Ok(_) => Ok(()),
+            Ok(_) => {
+                // Successfully sent to server
+                self.last_healthcheck = SystemTime::now();
```

Comment:
This is not the last health check time, this is the last time you talked to the server. You want to save this time when you actually health check the server in `pool.rs`.

Reply:
The variable name is misleading but this is how pgbouncer does these health checks. If I am reading it correctly, it is based on the server latest activity. If the server hasn't had recent activity, it gets checked.

https://github.com/pgbouncer/pgbouncer/blob/5c8a7728f79cf37f5fb040f49b327dbe8908d912/include/bouncer.h#L416

https://github.com/pgbouncer/pgbouncer/blob/aafe242fb8fcab38ccd67194d13ea836412f2af8/src/janitor.c#L149-L155

So perhaps we can just rename this to last_activity?

Reply:
Good catch.

Reply:
yeah originally named it this `latest_successful_server_interaction_timestamp` which admittedly is long, but changed it based on https://github.com/levkk/pgcat/pull/118#discussion_r942592349

Reply:
`last_activity` sounds good to me.